helm/prometheus-rules/templates/platform/atlas/alerting-rules/mimir.rules.yml

```diff
@@ -158,8 +158,7 @@ spec:
         dashboard: 09a5c49e9cdb2f2b24c6d184574a07fd/mimir-compactor-resources
         description: 'Mimir compactor has been failing its compactions for 2 hours.'
         opsrecipe: mimir#mimircompactorfailedcompaction
-      # Query is based on the following upstream mixin alerting rule : https://github.com/grafana/mimir/blob/main/operations/mimir-mixin-compiled/alerts.yaml#L858
-      expr: sum(increase(cortex_compactor_runs_failed_total{reason!="shutdown"}[2h])) by (cluster_id, installation, namespace, pipeline, provider) > 2
+      expr: min by (cluster_id, installation, namespace, provider, pipeline) (time() - (cortex_compactor_last_successful_run_timestamp_seconds)) > 60 * 60 * 2
```

Comment:
Any reason why you removed the comment?

Reply:
Because it's not based on the upstream mixin alerting rule anymore. So the comment does not make sense anymore.

Reply:
Sure, i'm tired 😅
Anyway, as you say, the alert does not fire if the value is 0 and that's why they need to have the `since startup` alert in both Loki and Mimir

Reply:
Oooh, I finally remember this 0 condition and why the unit tests work! Because `time()` starts at 0 for the tests! :facepalm: 
I'll fix it then.

Reply:
Should be fixed :grin:

Reply:
Oh yes that time function is tricky